src/java.base/share/classes/java/lang/System.java

```diff
@@ -732,6 +732,16 @@ public static native void arraycopy(Object src,  int  srcPos,
      *     <td>Java Runtime Environment specification version, whose value is
      *     the {@linkplain Runtime.Version#feature feature} element of the
      *     {@linkplain Runtime#version() runtime version}</td></tr>
+     * <tr><th scope="row">{@systemProperty java.specification.maintenance.version}</th>
+     *     <td>Java Runtime Environment specification maintenance version, defined only
+     *         when the specification implemented by this runtime has undergone a
+     *         <a href="https://jcp.org/en/procedures/jcp2#3.6.4">maintenance
+     *         release</a> <em>(optional)</em>.
+     *         After a first maintenance release this property will
+     *         have the value {@code "1"}; after a second maintenance
+     *         release, this property will have the value {@code "2"},
+     *         and so on.
```

Comment:
There should be no requirement that values be allocated sequentially.  In other words, if JCP MR <x> does not require an RI, then it should not be surprising if there is no JDK build with maintenance version <x>.  As an example, JSR 337 MR 1 and MR 2 both used the same RI.  If this system property had existed during development of MR 1, it would return "1".  Since no RI was submitted for MR 2, a build returning "2" should never exist.  MR 3 did update the RI, so it would return "3".

Reply:
> > Is there intent to backport this to the RI for JSR 337 MR 4?
> 
> I think that would be helpful, if not strictly necessary.

PS And if backported, the spec could be updated for earlier release trains to start at a different value like "4".

Reply:
In the latest push, to address the concerns raised updated the proposed wording to, in plain text:

Java Runtime Environment specification maintenance version, not defined before the specification implemented by this runtime has undergone a maintenance release (optional). When defined, the value of this property may be interpreted as a positive integer. The value indicates the latest maintenance release the runtime is known to support. A later release may be supported by the environment. To indicate the first maintenance release this property will have the value "1"; to indicate the second maintenance release, this property will have the value "2", and so on.

Reply:
PS CSR Updated to reflect this push; please review: https://bugs.openjdk.java.net/browse/JDK-8285764

Reply:
The negative definition above permits the property always to be undefined, but we do want it to be defined when meaningful. It’s also getting to be an awful lot of text to add to the otherwise terse tabular summary of system properties in the [`System::getProperties()` specification](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/System.html#getProperties()).

Consider using this text in the table:

> Java Runtime Environment specification maintenance version, which may be interpreted as a positive integer (optional, see below)

and then this in a paragraph following the table:

> The {@code java.specification.maintenance.version} property is defined if the specification implemented by this runtime at the time of its construction had undergone a <a href="https://jcp.org/en/procedures/jcp2#3.6.4">maintenance release</a>. When defined, its value identifies that maintenance release. To indicate the first maintenance release this property will have the value {@code "1"}; to indicate the second maintenance release this property will have the value {@code "2"}, and so on.